nginx/proxy.conf

```diff
@@ -11,4 +13,4 @@ proxy_set_header x-amz-date $now;
 proxy_set_header Authorization "AWS $aws_access_key:$aws_signature";
 proxy_set_header Host $bucket.s3.amazonaws.com;
 rewrite .* $key break;
-proxy_pass https://s3.amazonaws.com;
+proxy_pass https://$bucket.s3-$region.amazonaws.com;
```

Comment:
Tested for the default region, or us-east-1, it turns out default region url is special from the others ( right, since it is the first AWS region, lots about this region is special :) , base on this page, http://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region, and my test, only s3.amazonaws.com
 and s3-external-1.amazonaws.com works. I guess we can either use "external-1" as $region for us-east-1, and document it clearly!

Also, we can just use https://s3-$region.amazonaws.com, instead of https://$bucket.s3-$region.amazonaws.com

Sounds good?